crates/bevy_app/src/main_schedule.rs

```diff
@@ -15,6 +15,12 @@ use bevy_ecs::{
 /// By default, it will run the following schedules in the given order:
 ///
 /// On the first run of the schedule (and only on the first run), it will run:
+/// * [`StateTransition`]
+///      * This means that [`OnEnter(ExampleState)`] will be called *before* [`PreStartup`]
+///        if `ExampleState` is the starting state of the app.
```

Comment:
```suggestion
///        if `ExampleState` was added to the app with that initial state.
```
"starting state of the app" is confusing because apps are not associated with one global state type.

Reply:
Hmm. I like your suggestion, but I don't think this wording is correct, too, because of computed states: if a computed state happens to be derived from the initial enum variant of a state, its `OnEnter` would run here, too. Do you have something on your mind that will both capture this case and make the sentence not confusing?

Reply:
I think this is important to highlight here, as it can be a point of confusion. In our game, for example, we make use of `SubStates` and `ComputedStates` extremely heavily, I assume dozens of `ComputedStates` by the moment all features will be implemented - so most of the times it is `OnEnter(SomeComputedState)`.

Reply:
Sure, maybe change `ExampleState` to `MyState::Foo` and then say "if `MyState::Foo` is the initial value of `MyState`" or similar.